draft-ietf-rtcweb-mdns-ice-candidates.md

```diff
@@ -195,8 +195,11 @@ This procedure ensures that an mDNS name is used to replace only one IP address.
 Specifically, an ICE agent using an interface with both IPv4 and IPv6 addresses
 MUST expose a different mDNS name for each address.
 
-Any server-reflexive candidates generated from an mDNS local candidate MUST have
-their raddr field set to 0.0.0.0 and their rport field set to 0.
+Note that any server-reflexive candidates generated from an mDNS host
+candidate will have IP addresses as their transport addresses, rather than mDNS
+hostnames. Accordingly, they MUST NOT be considered redundant as discussed in  
+{{RFC8445}}. To avoid accidental leakage, these candidates MUST have their
```

Comment:
Can we resume the redundancy check after resolving an/all mDNS name(s)?

Reply:
IIUC, the issue might be that we would discard the srflx candidate as redundant against the mdns candidate. On the other hand, discarding the mdns candidate in favor of the srflx candidate is probably fine. It also has the benefit of skipping a potentially unsuccessful mdns resolution.

Reply:
@tQsW, what did you mean?

@youennf, yes, this is the issue I am trying to address. But the problem is that 8445 says that the lower-priority candidate (the srflx) SHOULD be eliminated, so eliminating the mDNS candidate isn't really an option.

Reply:
ok. A user agent might still decide to not expose the mdns candidate if not found useful as a potential optimization but filtering is done at a different level.
As @jeroendeb mentioned, not providing the mdns candidate gives a bit of information so always providing both seems good as a default.

Reply:
If this paragraph meant to apply to the remote side that receives the mDNS candidates, the redundancy elimination procedure in 8445 could still apply to reduce the connectivity checks after the name is resolved.

Reply:
LGTM.
As an alternate way of fixing this, we could state that, in the case of two redundant candidates as defined by RFC8445, neither of them should be eliminated if one is SRFLX and the other mDNS host.
That would only precise RFC8445 rule "The agent SHOULD eliminate the redundant candidate with the lower priority" instead of changing RFC8445 redundancy definition, thus only addressing the problem we know of.

Reply:
We're not actually changing the redundancy definition here, just clarifying to avoid confusion. I would certainly argue that a FQDN and literal IP are not equivalent/redundant.

As noted in the document, the ICE agent may decide not to wrap candidates that it knows are public, which seems like the right way to handle this case when the ICE agent has a priori knowledge. (Otherwise, the srflx candidate comes too late, after the mdns candidate has already been sent off).